### PR TITLE
Logical solver

### PR DESCRIPTION
**Pull request by Mike-3**

### CODE REVIEW

File: lib/tools/games/logical_solver/logic/logical_solver.dart [hunk 1]
```diff
@@ -0,0 +1,686 @@
...
+					}
+				}
+			}
+			for (var itemBlock = 0; itemBlock < min(logical.logicalItems.length, logicalItems.length); itemBlock++) {
+				for (var item = 0; item < min(logical.logicalItems[itemBlock].length, logicalItems[itemBlock].length); item++) {
+					logicalItems[itemBlock][item] = logical.logicalItems[itemBlock][item];
+				}
+			}
+		} else if (categoriesCount == 4 && itemsCount == 5) {
+			logicalItems[0][0] = 'Steffi';
```
Comment (S-Man42) [inline lib/tools/games/logical_solver/logic/logical_solver.dart:235]: Why are here values?

**Comment by S-Man42** [comment]
Please rename to Logical Supporter
Missing tests

**Comment by S-Man42** [comment]
Pls add sources to registry

File: lib/tools/games/logical_solver/logic/logical_solver.dart (hunk 1 above)
Comment (Mike-3) [reply lib/tools/games/logical_solver/logic/logical_solver.dart:235]: Init/ default logical

**Comment by Mike-3** [comment]
Without references/ sources

**Comment by Mike-3** [comment]
done

**Comment by S-Man42** [comment]
Resolve conflicts

**Comment by Mike-3** [comment]
done

**Pull request merged**